Added checkPackageInstalled and saveDeAsJson

rrydbirk opened the pull request:
As agreed in https://github.com/kharchenkolab/cacoa/issues/34, I added checkPackageInstalled. Further, I added saveDeAsJson so it can be removed from Cacoa and Conos, and I included deview files for this. Finally, I included a bugfix for dotPlot.

rrydbirk committed:
Added checkPackageInstalled

rrydbirk committed:
Added saveDeAsJson

rrydbirk committed:
Updated imports

rrydbirk committed:
Update and bugfix for dotPlot

rrydbirk committed:
Defined global variables

rrydbirk committed:
Updated documentation

rrydbirk committed:
Updated CHANGELOG

VPetukhov commented on R/plots.R:
```diff
@@ -1,14 +1,15 @@
 #' @importFrom grDevices adjustcolor rainbow colorRampPalette
 #' @importFrom magrittr %>% %<>% %$%
 #' @importFrom rlang .data
-#' @import scales
```
Are you sure it's never used?

VPetukhov commented on R/saveDeAsJson.R:
```diff
@@ -0,0 +1,118 @@
+#' @export
+saveDeAsJson <- function(de.raw, sample.groups=NULL, saveprefix=NULL, dir.name="JSON", gene.metadata=NULL,verbose=TRUE) {
+
+  checkPackageInstalled("jsonlite", cran=TRUE)
```
If you use it here, it should be in Suggests :)

VPetukhov commented on R/saveDeAsJson.R:
```diff
@@ -0,0 +1,118 @@
+  write(s, file=toc.file)
+
+  # Copy viewer files
+  file.copy(from = c(paste(system.file(package="sccore"), "extdata/deview.2.html", sep ="/"),
+                     paste(system.file(package="sccore"), "extdata/deview.2.js", sep ="/")), to = paste0(dir.name), overwrite = TRUE)
```
I'd strongly suspect this not to work on Windows because of "/" in the path... In any case, the [recommended way](https://r-pkgs.org/data.html#filepaths) `system.file("extdata", "deview.2.html", package="sccore")` would be shorter and safer.

VPetukhov replied:
Btw, it can also be vectorised as
```r
system.file("extdata", c("deview.2.html", "deview.2.js"), package="sccore")
```
:)

evanbiederstedt replied:
Could we make these `deview2.html` and `deview.2.js` please?

rrydbirk replied:
@evanbiederstedt, what do you mean by "making these", writing them from scratch each time? I've uploaded them in [13400ff](https://github.com/kharchenkolab/sccore/commit/13400ff5a32b054a1dc6096079c03cd83a3d68c5)

rrydbirk replied:
Added Viktor's suggestion

VPetukhov reviewed (changes requested):
Looks good to me, thank you @rrydbirk! Needs a couple of small fixes for saveDeAsJson.R and that's it.

evanbiederstedt commented on R/helpers.R:
```diff
@@ -164,3 +164,40 @@ mergeCountMatrices <- function(cms, transposed=FALSE, ...) {
+#' @param bioc logical Package(s) is/are available from Bioconductor (default = FALSE)
+#' @param cran logical Package(s) is/are available from CRAN (default = FALSE)
+#' @examples
+#' checkPackageInstalled("sccore", cran = TRUE)
```
This example will probably fail CRAN upon upload....

We could try dontrun as well, but I remember running into issues with this

https://stackoverflow.com/questions/12038160/how-to-not-run-an-example-using-roxygen2

It's probably best just to remove the example (sadly)

rrydbirk replied:
Could checkPackageInstalled("base") work? Otherwise, let's at least go for dontrun

evanbiederstedt replied:
Let's try `dontrun`

If there's a problem, I'll just remove the example

evanbiederstedt commented on R/helpers.R:
```diff
@@ -164,3 +164,40 @@ mergeCountMatrices <- function(cms, transposed=FALSE, ...) {
   return(res)
 }
+
+#' Check for installed package(s)
```
I'd remove this line

Do the following:

```
#'  Check whether a package is installed and suggest how to install from either CRAN or Bioconductor
#'
#' @param pkgs character Package name(s)
```

You'll see that the roxygen2 docs will render correctly

rrydbirk replied:
Implemented

evanbiederstedt commented on R/plots.R:
```diff
@@ -522,7 +523,7 @@ dotPlot <- function (markers,
                      scale.min = NA,
                      scale.max = NA,
-                     verbose=TRUE,
+                     verbose=FALSE,
```
Why was this changed?

rrydbirk replied:
Honestly, I introduced the progress bar at a time where I was in love with these things. However, in my experience this function is quite fast, no need to show bars per default

evanbiederstedt commented:
@rrydbirk 

(1) Please increase the version in the DESCRIPTION and README as 1.0.3

(2) It's best not to make changes directly in `dev`, but make a branch off of `dev` and do a PR against this. 

I only push to `main` is there's a new version on CRAN; I then release that version

rrydbirk commented:
Thanks @VPetukhov and @evanbiederstedt for review.
About branches, I apologize for my misdemeanor, I'm still not that experienced with this. Should I push the suggested changes to `dev` this time, or...?

rrydbirk committed:
Bumped version, added 'jsonlite' to suggests

rrydbirk committed:
Edited 'checkPackageInstalled' and 'saveDeAsJson' according to merge comments

rrydbirk committed:
Updated documentation

rrydbirk committed:
Resolved merge conflicts

Merge branch 'main' of github.com:kharchenkolab/sccore into dev

# Conflicts:
#	CHANGELOG.md
#	README.md

evanbiederstedt reviewed (approved):
LGTM, 
I'll include in the new version. 

I'll close the PR though as I might need to make changes to dev for the release
